commit a7ee17c52e67f2df445abf760854afc534987ae6
Author: Michael Biebl <biebl@debian.org>
Date:   Fri Feb 25 01:57:50 2011 +0100

    Revert "plugins/evolution: Making it work with current jhbuild of Evolution"
    
    This reverts commit 5a48b5c8d339c5eed04e34e9c672d1925352fb28.
    
    Revert "plugins/evolution: Fixes for making it work on latest master"
    
    This reverts commit 87a5064e3033e1521b3ba01c63a5667bde231a82.

diff --git a/configure.ac b/configure.ac
index 48c1989..f447f53 100644
--- a/configure.ac
+++ b/configure.ac
@@ -186,10 +186,9 @@ LIBFLAC_REQUIRED=1.2.1
 LIBEXIF_REQUIRED=0.6
 LIBGSF_REQUIRED=1.13
 EXEMPI_REQUIRED=2.1.0
-EVO_REQUIRED=2.91.90
-EVO_SHELL_REQUIRED=2.91.90
-EDS_REQUIRED=2.91.90
-CAMEL_REQUIRED=2.91.90
+EVO_REQUIRED=2.31.2
+EDS_REQUIRED=2.31.2
+CAMEL_REQUIRED=2.31.2
 # Unlikely version for now, Nepomuk integration isn't finished in streamanalyzer atm
 LIBSTREAMANALYZER_REQUIRED=0.7.0
 GEE_REQUIRED=0.3
@@ -353,7 +352,6 @@ TRACKER_MINER_FS_LIBS="$TRACKER_MINER_FS_LIBS -lz -lm"
 evolution_plugin_name="evolution-plugin"
 
 TRACKER_MINER_EVOLUTION_REQUIRED="glib-2.0                  >= $GLIB_REQUIRED
-                                  evolution-shell           >= $EVO_SHELL_REQUIRED
                                   evolution-plugin          >= $EVO_REQUIRED
                                   evolution-data-server-1.2 >= $EDS_REQUIRED
                                   camel-1.2                 >= $CAMEL_REQUIRED"
diff --git a/src/plugins/evolution/tracker-evolution-plugin.c b/src/plugins/evolution/tracker-evolution-plugin.c
index c12a237..c6f1a87 100644
--- a/src/plugins/evolution/tracker-evolution-plugin.c
+++ b/src/plugins/evolution/tracker-evolution-plugin.c
@@ -43,13 +43,10 @@
 #include <camel/camel.h>
 
 #include <mail/mail-config.h>
+#include <mail/mail-session.h>
 #include <mail/em-utils.h>
 #include <mail/mail-ops.h>
 
-#include <mail/e-mail-session.h>
-#include <mail/e-mail-backend.h>
-#include <shell/e-shell.h>
-
 #include <e-util/e-config.h>
 #include <e-util/e-account-utils.h>
 
@@ -186,7 +183,6 @@ static TrackerEvolutionPlugin *manager = NULL;
 static GStaticRecMutex glock = G_STATIC_REC_MUTEX_INIT;
 static guint register_count = 0, walk_count = 0;
 static ThreadPool *folder_pool = NULL;
-static EMailSession *session = NULL;
 
 /* Prototype declarations */
 static void register_account (TrackerEvolutionPlugin *self, EAccount *account);
@@ -1250,10 +1246,7 @@ get_last_deleted_time (TrackerEvolutionPlugin *self)
 				continue;
 			}
 
-			if (!(store = (CamelStore *) camel_session_get_service (CAMEL_SESSION (session),
-			                                                        uri,
-			                                                        CAMEL_PROVIDER_STORE,
-			                                                        NULL))) {
+			if (!(store = (CamelStore *) camel_session_get_service (session, uri, CAMEL_PROVIDER_STORE, NULL))) {
 				continue;
 			}
 
@@ -1364,11 +1357,7 @@ register_walk_folders_in_folder (TrackerEvolutionPlugin *self,
 		/* This is asynchronous and hooked to the mail/ API, so nicely
 		 * integrated with the Evolution UI application */
 
-		mail_get_folder (session,
-		                 iter->uri,
-		                 0,
-		                 register_on_get_folder,
-		                 info,
+		mail_get_folder (iter->uri, 0, register_on_get_folder, info,
 		                 mail_msg_unordered_push);
 
 		if (iter->child) {
@@ -1438,11 +1427,7 @@ unregister_walk_folders_in_folder (TrackerEvolutionPlugin *self,
 		/* This is asynchronous and hooked to the mail/ API, so nicely
 		 * integrated with the Evolution UI application */
 
-		mail_get_folder (session,
-		                 titer->uri,
-		                 0,
-		                 unregister_on_get_folder,
-		                 info,
+		mail_get_folder (titer->uri, 0, unregister_on_get_folder, info,
 		                 mail_msg_unordered_push);
 
 		if (titer->child) {
@@ -1603,10 +1588,7 @@ introduce_account_to (TrackerEvolutionPlugin *self,
 	if (!(provider->flags & CAMEL_PROVIDER_IS_STORAGE))
 		return;
 
-	if (!(store = (CamelStore *) camel_session_get_service (CAMEL_SESSION (session),
-	                                                        uri,
-	                                                        CAMEL_PROVIDER_STORE,
-	                                                        NULL))) {
+	if (!(store = (CamelStore *) camel_session_get_service (session, uri, CAMEL_PROVIDER_STORE, NULL))) {
 		return;
 	}
 
@@ -1902,10 +1884,7 @@ register_account (TrackerEvolutionPlugin *self,
 	if (!(provider->flags & CAMEL_PROVIDER_IS_STORAGE))
 		return;
 
-	if (!(store = (CamelStore *) camel_session_get_service (CAMEL_SESSION (session),
-	                                                        uri,
-	                                                        CAMEL_PROVIDER_STORE,
-	                                                        NULL))) {
+	if (!(store = (CamelStore *) camel_session_get_service (session, uri, CAMEL_PROVIDER_STORE, NULL))) {
 		return;
 	}
 
@@ -1970,10 +1949,7 @@ unregister_account (TrackerEvolutionPlugin *self,
 	if (!(provider->flags & CAMEL_PROVIDER_IS_STORAGE))
 		return;
 
-	if (!(store = (CamelStore *) camel_session_get_service (CAMEL_SESSION (session),
-	                                                        uri,
-	                                                        CAMEL_PROVIDER_STORE,
-	                                                        NULL))) {
+	if (!(store = (CamelStore *) camel_session_get_service (session, uri, CAMEL_PROVIDER_STORE, NULL))) {
 		return;
 	}
 
@@ -2194,15 +2170,6 @@ tracker_evolution_plugin_init (TrackerEvolutionPlugin *plugin)
 	TrackerEvolutionPluginPrivate *priv = TRACKER_EVOLUTION_PLUGIN_GET_PRIVATE (plugin);
 	EIterator *it;
 
-	if (!session) {
-		EShell *shell;
-		EShellBackend *shell_backend;
-
-		shell = e_shell_get_default ();
-		shell_backend = e_shell_get_backend_by_name (shell, "mail");
-		session = e_mail_backend_get_session (E_MAIL_BACKEND (shell_backend));
-	}
-
 	priv->connection = NULL;
 	priv->last_time = 0;
 	priv->resuming = FALSE;
